### DISCUSSION TITLE
Error: Can't output Elems with invalid ids!

### DISCUSSION THREAD

**Discussion by Ali-toghraee** [idaholab/moose#17158] [Q&A Modules: General] [answered]
Hello MOOSE users,

My model has Phase Field+ Tensor mechanics kernels and I am using the combined module. 
When my Tensor mechanic is Elastic ( elastic stress and strain) I have no problem. 
But, when I add plasticity to my model , I get this error of **"Can't output Elems with invalid ids!"** after some steps. 

What I figured out so far is that it is some how related to the adaptive mesh I have. For example, if I chose to update my mesh every 6 iterations, the error occurs around t = 5e-6 sec. But, if I update every 18 iterations it lasts up to  t = 5e-4 sec. 
I would appreciate any help on this matter.

I have attached some part of my input file.
[plastic_github.txt](https://github.com/idaholab/moose/files/6059093/plastic_github.txt)

P.S. My MOOSE version is kinda old!

**Comment by aeslaughter** [comment]
@sapitts Have you seen this before with Tensor Mechanics?

**Comment by sapitts** [comment, marked as answer]
I have not seen this error before, but based on some discussions I've had with @laagesen and @dschwen I believe there is an issue with plasticity stateful materials and mesh adaptivity: we don't currently have the infrastructure to handle moving over stateful properties (e.g. the old plastic state from the previous timestep) to a modified mesh. Daniel or Larry, would one of you be willing to clarify/ expand on this issue please?

It could be that you see the difference in the "allowable" simulation time is a function of both the onset of plasticity and how close to that onset the mesh is modified through adaptivity. I suspect that as soon as the simulation goes to look for the old plastic state material property information on a mesh that has been modified by adaptivity. If you delay adaptivity, the simulation should be able to run longer.

**Reply by Ali-toghraee** [reply]
Hi sapitts,

Thanks for the explanation. You are right about the delay for adaptivity . when I set the adaptivity  from interval = 6 to interval = 36, my simulations runs for very longer time. 

Is there a way to define a fixed refined mesh at certain areas in the domain? I am not sure if the Pattern Mesh can do the trick.

Alireza

**Reply by hugary1995** [reply]
You could use initial adaptivity. I'm pretty sure that'll solve your issue here.

**Reply by hugary1995** [reply]
Take a look at this input file for example: [https://github.com/idaholab/moose/blob/next/test/tests/adaptivity/initial_adapt/initial_adapt.i](https://github.com/idaholab/moose/blob/next/test/tests/adaptivity/initial_adapt/initial_adapt.i) . But as Stephanie said, incremental formulations in tensor mechanics with adaptivity are still under the works.

**Reply by Ali-toghraee** [reply]
Thanks hungry, It helped a lot. Do you know how can I define multiple boxes instead of one?

**Reply by hugary1995** [reply]
Combo marker is for that exact purpose: [https://mooseframework.inl.gov/source/markers/ComboMarker.html](https://mooseframework.inl.gov/source/markers/ComboMarker.html)

**Reply by Ali-toghraee** [reply]
Thank you , yep found it 10 min ago :)
